packages/core/src/actions/execute.ts

```diff
@@ -321,24 +475,90 @@ export const approveDeploymentViaSigner: ApproveDeployment = async (
   approvalLeafWithProof,
   executionMode,
   provider,
-  signer
+  ownerSignatures,
+  deploymentContext
 ) => {
-  const ownerSignature = await signMerkleRoot(merkleRoot, signer)
+  const sphinxModule = new ethers.Contract(moduleAddress, SphinxModuleABI)
 
-  const sphinxModule = new ethers.Contract(
-    moduleAddress,
-    SphinxModuleABI,
-    signer
+  const packedOwnerSignatures = ethers.solidityPacked(
+    new Array(ownerSignatures.length).fill('bytes'),
+    ownerSignatures
+  )
+  const approvalData = sphinxModule.interface.encodeFunctionData('approve', [
+    merkleRoot,
+    approvalLeafWithProof,
+    packedOwnerSignatures,
+  ])
+
+  return deploymentContext.executeTransaction(
+    deploymentContext,
+    {
+      to: moduleAddress,
+      data: approvalData,
+      chainId: (await provider.getNetwork()).chainId.toString(),
+    },
+    executionMode
   )
+}
 
-  return (
-    await sphinxModule.approve(
-      merkleRoot,
-      approvalLeafWithProof,
-      ownerSignature,
-      await getGasPriceOverrides(provider, signer, executionMode)
+export const executeTransactionViaSigner: ExecuteTransaction = async (
+  deploymentContext: DeploymentContext,
+  transaction: MinimumTransaction,
+  executionMode: ExecutionMode
+): Promise<ethers.TransactionReceipt> => {
```

Comment:
Used for sending transactions via an ethers signer in the deploy task and proposal simulation.